tests/lib/cpp/cxx/src/main.cpp

```diff
@@ -153,3 +153,12 @@ DEVICE_DT_DEFINE(DT_NODELABEL(test_dev1_dfr), NULL, NULL, NULL, NULL, POST_KERNE
 static int fake_pm_action(const struct device *dev,
 		enum pm_device_action pm_action) { return -1; }
 PM_DEVICE_DT_DEFINE(DT_NODELABEL(test_dev0_boot), fake_pm_action);
+
+static int test_pm_device(void)
+{
+	/* Reference the data structure to avoid an "unused-variable" warning */
+	(void)PM_DEVICE_DT_GET(DT_NODELABEL(test_dev0_boot));
```

Comment:
Don't we have something like a macro/attribute to inform the compiler to ignore the unused checks? This function that does nothing but cast is... something.

Reply:
of course but we can't really apply it in this case due to how `PM_DEVICE_DT_DEFINE` expands

Reply:
I don't get how this actually does anything useful, to me it looks like the compiler is complaining and that now results in us creating this bodge to make it happy. That really feels wrong on some guttural level to me.

Reply:
Sure, what alternative are you proposing?

Reply:
I guess avoiding C++ warnings that conflict with C warnings somehow? I don't know C++ that well anymore but there's like a 100 warning flags for the compiler and tons of attributes and such, none of them help us avoid this sort of thing?

Reply:
maybe I can instantiate a device instead of a dummy function with a void reference, hang on

Reply:
done, makes a lot more sense honestly